runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -177,9 +177,9 @@ public void testDefaultInvalidGcpTempLocation() {
   @Test
   public void testDefaultStagingLocationUnset() {
     DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    options.setProject("");
     thrown.expect(IllegalArgumentException.class);
-    thrown.expectMessage(
-        "Error constructing default value for stagingLocation: gcpTempLocation is missing.");
+    thrown.expectMessage("--project is a required option.");
     options.getStagingLocation();
   }
```

Comment:
I added the test I think is described above, which indeed produces a null gcpTempLocation.

Going forward, I'm unclear on whether we can easily solve this issue in this PR.  The most expedient option seems like it would be to move the default bucket code to gcpTempLocation, but I imagine there are downsides.